lib/internal/modules/esm/loader.js

```diff
@@ -237,13 +251,60 @@ class DefaultModuleLoader {
    * @param {string} parentURL Path of the parent importing the module.
    * @param {Record<string, string>} importAssertions Validations for the
    *                                                  module import.
-   * @returns {Promise<ExportedHooks | KeyedExports[]>}
```

Comment:
For the sake of simplifying my life, can you include the added line as well for this kind of comments? It's a bit hard to comment without having the context and for some reason the GitHub UI only shows the line above.

The JSDoc is wrong/outdated, loader.import never returns an array – it used to before https://github.com/nodejs/node/pull/48296 and I guess we forgot about JSDoc in that PR. `ExportedHooks` doesn't seem to fit very well here either, if you prefer we can address that in its own PR.

Reply:
I think `ModuleExports` is correct (`import` is not limited to hooks).

I think `KeyedExports` is obsolete as of https://github.com/nodejs/node/pull/48296: import previously returned `{ [moduleUrl]: ModuleExports }` in order for the caller to know what exports belonged to which module (because multiple were possible).